src/components/books/BooksForm.js

```diff
@@ -1,21 +1,36 @@
-import React, { useState } from 'react';
-import { useDispatch } from 'react-redux';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { nanoid } from '@reduxjs/toolkit';
-import { addBook } from '../../redux/book-redux/BookSlice';
 
-const BooksForm = () => {
+/* Important please do not remove any eslint disables as it hepls me to write
+ "item_id" using underscrore  which is otherwise not possible.
+ Using camelcase will throw an error. It does not send proper request to the API.
+ */
+
+const BooksForm = ({ onAddBook }) => {
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
-  const dispatch = useDispatch();
+  const [category, setCategory] = useState('fiction');
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    const id = nanoid();
-    dispatch(addBook({ id, title, author }));
+    // eslint-disable-next-line camelcase
+    const item_id = nanoid();
+    onAddBook({
+      // eslint-disable-next-line camelcase
```

Comment:
- [ ] Good job so far, but note that it is not a good practice to disable linters in your code because linters force you to write bug-free and clean code. Please remove all areas you disabled linters in your app.

 I guess the item_id made you disable linters. To fix this issue, you can reassign the data coming from the api to your own state so that you will be able to use the variable name of your choice.

e.g

in your reducer, you can do (this is just an example) :

```
{
"itemId": item_id,
title,
author
}
```